functors task (1)/student/include/impl.h

```diff
@@ -1,13 +1,18 @@
 #pragma once
 #include "bar_serving.h"
 #include <functional>
+#include <map>
 
 /**
  * @todo Implement functor-generator that will return next beer brand (cyclic)
  * @example HoeGaarden -> Corona -> Carlsberg -> Bud -> ZlataPraha -> Leffe -> HoeGaarden...
  */
 struct BeerOrganizer
 {
+    BeerBrand operator()() const;
+
+    private:
+    mutable int8_t counter = static_cast<int8_t>(BeerBrand::None);
```

Comment:
_prefer size_t_
why mutable?

Reply:
mutable because I declare BeerBrand operator()() as const method

Reply:
I think mutable is not a good solution here.
In fact, you modify the internal state, so it shouldn't be const.

Reply:
Done. Its make sense, thank.